### ISSUE TITLE
Feature request: Allow all descendant elements of configured `allowed_elements`

### ISSUE THREAD

**Issue by sideshowbarker** [DavidAnson/markdownlint#419]
For any element named in the MD033 `allowed_elements`, allow any child elements of the named elements.

Use case: To cause all HTML tables to be allowed, I have the following configured:

```yaml
allowed_elements: ["table", "tr", "td", "th", "thead", "tbody", "tfoot", "caption", "colgroup", "col"]
```
However, the actual tables in my documents also contain descendant `<code>…</code>`, and `<p>…</p>`, etc., markup.

I want markdownlint to report those `<code>…</code>`, and `<p>…</p>`, etc., elements when they occur anywhere else in a document — but I just don’t want markdownlint to report them when they are descendants of an HTML `table` element.

So if this feature were implemented, everything would work if I just did the following:

```yaml
allowed_elements: ["table"]
```

That is, the above would allow any HTML `table` and all its descendant elements.
